src/AutoRest.CSharp/Common/Output/Models/Types/TypeProvider.cs

```diff
@@ -15,6 +15,9 @@ namespace AutoRest.CSharp.Output.Models.Types
     internal abstract class TypeProvider
     {
         private readonly Lazy<INamedTypeSymbol?> _existingType;
+        private readonly Lazy<INamedTypeSymbol?> _contractType;
```

Comment:
I don't feel like it's better to add such many additional fields/properties to the base `TypeProvider`, this affects every derived type since it is the base type of generator output types.

I tried to use `SignatureType` to hold the breakingchange overload before, it isolates the functionality in a central place.